lib/rails_param/validator.rb

```diff
@@ -0,0 +1,40 @@
+module RailsParam
+  module Param
+    class Validator
+      attr_reader :parameter, :name, :options, :value
+
+      def initialize(parameter)
+        @parameter = parameter
+        @name = parameter.name
+        @options = parameter.options
+        @value = parameter.value
+      end
+
+      def validate!
+        options.each_key do |key|
+          klass = camelize(key)
+          Validator.const_get(klass).new(parameter).valid!
+        end
+      end
+
+      def valid!
+        raise InvalidParameterError, error_message unless valid_value?
+      end
+
+      private
+      def camelize(term)
```

Comment:
It's a shame we can't just use the active support `String#camelize` here... Maybe in the future (I think it's already a dev dependency and this gem depends on rails anyway).